etc/scripts/json2csv.py

```diff
@@ -106,239 +75,148 @@ def load_scan(json_input):
+                    inf = OrderedDict(Resource=path)
+                    inf[header] = cop
+                    inf['start_line'] = start_line
+                    inf['end_line'] = end_line
+                    yield inf
+
+        for email in scanned_file.get('emails', []):
+            email_info = OrderedDict(Resource=path)
             for k, val in email.items():
-                email[k] = val
-                keys.add(key)
-            rows.append(email)
-
-
-    # keys = ['Resource'] + [x for x in keys if x != 'Resource']
-    all_rows = []
-    for row in rows:
-        d = OrderedDict()
-        for k in keys.keys():
-            d[k] = row.get(k, '')
-        all_rows.append(d)
+                email_info[k] = val
+            yield email_info
 
-    return all_rows
+        for url in scanned_file.get('urls', []):
+            url_info = OrderedDict(Resource=path)
+            for k, val in url.items():
+                url_info[k] = val
+            yield url_info
+
+        excluded_columns = ('packaging',
+                            'payload_type',
+                            'keywords_doc_url',
+                            'download_sha1',
+                            'download_sha256',
+                            'download_md5',
+                            'code_view_url',
+                            'vcs_tool',
+                            'vcs_revision',
+                            'license_expression')
+
+        for package in scanned_file.get('packages', []):
+            pack = OrderedDict(Resource=path)
+            for k, val in package.items():
+                nk = 'package__' + k
+                if not isinstance(val, (list, dict, OrderedDict)):
+                    if k not in excluded_columns:
+                        pack[nk] = val
+                elif k in ('authors', 'download_urls', 'copyrights', 'asserted_licenses'):
+                    if len(val) > 0:
+                        if k == 'authors':
+                            # We only want the first author
+                            pack[nk] = val[0]['name']
+                        if k == 'download_urls':
+                            # We only want the first URL
+                            pack[nk] = val[0]
+                        if k == 'copyrights':
+                            # All copyright statements are to be joined in a multi-line cell
+                            pack[nk] = '\n'.join(val)
+                        if k == 'asserted_licenses':
+                            # All license names are to be joined in a multi-line cell
+                            licenses = [license_info['license'] for license_info in val]
+                            pack[nk] = '\n'.join(licenses)
+                    else:
+                        pack[nk] = ''
+            yield pack
+
+
+def collect_header_keys(scan_data):
+    """
+    Return a list of keys collected from a list of scanned data dictionaries.
+    """
+    keys = OrderedSet()
```

Comment:
If this is the only place where you use an OrderedSet you could get rid of the OrderedSet class and fold the bits of code from it in that function for simplicity